temporalio/ext/src/client.rs

```diff
@@ -295,19 +296,13 @@ where
         async move { fut.await.map(|msg| msg.get_ref().encode_to_vec()) },
         move |ruby, result| {
             let block = ruby.get_inner(block);
-            match result {
-                Ok(val) => {
-                    // TODO(cretz): Any reasonable way to prevent byte copy?
-                    let _: Value = block
-                        .call((RString::from_slice(&val),))
-                        .expect("Block call failed");
-                }
-                Err(status) => {
-                    let _: Value = block
-                        .call((RpcFailure { status },))
-                        .expect("Block call failed");
-                }
+            let _: Value = match result {
+                // TODO(cretz): Any reasonable way to prevent byte copy that is just going to get decoded into proto
+                // object?
```

Comment:
There probably is, but it might also be more effort than it's worth. You'd have to pass around some explicit lifetime-having struct here or something. Hard for me to say without playing with the code myself but I could try it out if you like

Reply:
Yeah, I think I found a way with `rb_str_new_static` (which magnus doesn't expose directly, but uses for static literals in `r_string!`). but yeah it's a dangerous lifetime game to play save these cycles. I'll consider it premature optimization for now and we can come back later if needed.